### PR TITLE
Create generic items without Info()

### PR DESCRIPTION
**Pull request by ashmrtn**
Create generic Item implementations that
don't implement the ItemInfo interface.
These implementations can be used for
things like metadata files.

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #4191

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/data/item.go
Comment (ryanfkeepers) [inline src/internal/data/item.go:61]: The name here is getting a little chonky.  `Prefetched` wasn't entirely easy to grok, but worked out as we had either prefetched or non-prefetched items, so things could be extrapolated.  `UnindexedPrefetched` implies the possibility of an `IndexedPrefetched` item, and I don't know where that would get us.  Further, without any reference to an index elsewhere in this code, and no explanation in the comment, I'm not even sure what the indexing here is referring to.

I feel like this decomposition could use a re-do.  Or, at least, a renaming.  some thoughts:
```go
// basically UnindexedPrefetched
type BaseItem struct {
  id, reader, modTime
}

type ItemAndInfo struct {
  BaseItem, ItemInfo
}

type LazyItem struct {
  ItemAndInfo, Getter, Errs
}
```

Low effort go at it.  Thoughts?

**Review by ryanfkeepers — APPROVED** [review]
Technically workable, so approving to unblock, but I think the names need a review if we want concepts to remain maintainable.

File: src/internal/data/item.go
Comment (ashmrtn) [reply src/internal/data/item.go:61]: I'll come back and fix naming after I get the PR train merged

**Pull request merged**
